src/ert/_c_wrappers/config/config_parser.py

```diff
@@ -27,6 +40,77 @@ def __init__(self, errors: str, config_file: Optional[str] = None) -> None:
         )
 
 
+class SingleConfigError(ConfigValidationError):
+    def __init__(self, error: str, filename: str, location: Optional[Location] = None):
+        self.error = error
```

Comment:
This starting to look good! I just realized over the weekend that we probably could make `ConfigValidationError` have the following `__init__`:

```python
class ConfigValidationError(ValueError):
    def __init__(self, errors: str, location: Optional[Union[str,Location]] = None) -> None:
```
then we don't need to introduce `SingleConfigError`.

Reply:
I added this in now, removing SingleConfigError. I added (temporarily) some stuff to ConfigValidationError to support its future + current use, planning to remove the redundant stuff when the "old way" of calling/instantiating it is deprecated:

in ConfigValidationError: remove config_file getter/setter
in ConfigValidationError.init: remove config_file argument
in ConfigValidationError: take in only one error at a time